gstreamer-sdp/src/s_d_p_message.rs

```diff
@@ -1,11 +1,440 @@
+		unsafe {
+			from_glib(ffi::gst_sdp_message_set_connection(self.to_glib_none_mut().0, nettype.to_glib_none().0, addrtype.to_glib_none().0, address.to_glib_none().0, ttl, addr_number))
+		}
+	}
+
+	pub fn set_information(&mut self, information: &str) -> SDPResult {
+		unsafe {
+			from_glib(ffi::gst_sdp_message_set_information(self.to_glib_none_mut().0, information.to_glib_none().0))
+		}
+	}
+
+	pub fn set_key(&mut self, type_: &str, data: &str) -> SDPResult {
+		unsafe {
+			from_glib(ffi::gst_sdp_message_set_key(self.to_glib_none_mut().0, type_.to_glib_none().0, data.to_glib_none().0))
+		}
+	}
+
+    pub fn set_origin(&mut self, username: &str, sess_id: &str, sess_version: &str, nettype: &str, addrtype: &str, addr: &str) -> SDPResult {
+        unsafe {
+            from_glib(ffi::gst_sdp_message_set_origin(self.to_glib_none_mut().0, username.to_glib_none().0, sess_id.to_glib_none().0, sess_version.to_glib_none().0, nettype.to_glib_none().0, addrtype.to_glib_none().0, addr.to_glib_none().0))
+        }
+    }
+
+    pub fn set_session_name(&mut self, session_name: &str) -> SDPResult {
+        unsafe {
+            from_glib(ffi::gst_sdp_message_set_session_name(self.to_glib_none_mut().0, session_name.to_glib_none().0))
+        }
+    }
+
+    pub fn set_uri(&mut self, uri: &str) -> SDPResult {
+        unsafe {
+            from_glib(ffi::gst_sdp_message_set_uri(self.to_glib_none_mut().0, uri.to_glib_none().0))
+        }
+    }
+
+    pub fn set_version(&mut self, version: &str) -> SDPResult {
+        unsafe {
+            from_glib(ffi::gst_sdp_message_set_version(self.to_glib_none_mut().0, version.to_glib_none().0))
         }
     }
+
+    pub fn times_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_message_times_len(self.to_glib_none().0)
+        }
+    }
+
+    pub fn zones_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_message_zones_len(self.to_glib_none().0)
+        }
+    }
+
+    pub fn as_uri(scheme: &str, msg: &SDPMessage) -> Option<String> {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gst_sdp_message_as_uri(scheme.to_glib_none().0, msg.to_glib_none().0))
+        }
+    }
+
+    pub fn parse_buffer(data: &[u8]) -> Result<Self, SDPResult> {
+        assert_initialized_main_thread!();
+		unsafe {
+            let size = data.len() as u32;
+            let mut msg = glib_ffi::g_malloc(mem::size_of::<ffi::GstSDPMessage>());
+            let result = from_glib(ffi::gst_sdp_message_parse_buffer(data.to_glib_none().0, size, &mut msg));
+            match result {
+                SDPResult::ok => Some(from_glib_full(msg)),
+                _ => Err(result),
+            }
+		}
+    }
+
+    pub fn parse_uri(uri: &str, msg: &mut SDPMessage) -> Result<Self, SDPResult> {
+        assert_initialized_main_thread!();
+		unsafe {
+            let mut msg = glib_ffi::g_malloc(mem::size_of::<ffi::GstSDPMessage>());
```

Comment:
So in some places I saw usage of `std::mem::uninitialized` so I wasn't sure if I'm supposed to use that or glib's `g_malloc` method. Although, my operating assumption was that the compiler would infer the memory size to allocate?

Reply:
malloc is needed if it should be heap allocated and there is a Drop impl that frees the data (in all cases when you use the glib_wrapper macro), otherwise you can simply stack allocate it